python/lsst/daf/butler/registry/_registry.py

```diff
@@ -284,11 +284,11 @@ def __init__(self, database: Database, *,
             # store managers and their versions in attributes table
             context.addInitializer(lambda db: versions.storeManagersConfig())
             context.addInitializer(lambda db: versions.storeManagersVersions())
-            # dump universe config as yaml into attributes
+            # dump universe config as json into attributes (faster than YAML)
             # `dimCfg` is to make freaking mypy happy
```

Comment:
This is quite confusing to me because it means that mypy sees a path through the earlier code where dimensionConfig can still be None -- I can't see it though. We have an "if None" test above and in that branch we always set it or raise so it seems like a bug in mypy.  The odd thing is that it doesn't get upset by the line below that assigned dimensionConfig to dimCfg.

@TallJimbo can you see the problem?

Reply:
Maybe it gets confused by context manager which wraps assignment to `dimensionConfig`?

Reply:
I was confused when I saw it originally in the review, too, but I didn't see an obvious explanation.

Reply:
Context managers can potentially suppress exceptions inside `with` clause, so this is probably the path that `mypy` has to assume still exists (I presume it does not analyze how our context manager is implemented). I think I also tried to add `assert dimensionConfig is not None` after that context but it did not help with mypy.

Reply:
I fixed the mypy problem -- mainly by calling dump first and then copying the resulting Optional[str] to a str